### PR TITLE
Update whitepaper to use hash(previous block) for fill ID

### PR DESCRIPTION
**Pull request by cducrest**

### CODE REVIEW

File: docs/source/whitepaper.rst [hunk 1]
```diff
@@ -66,6 +65,7 @@ The parameters that need to be specified by the users are:
 - recipient address on target rollup
 - amount
 - validity period of request
+- expiry block number
```
Comment (fredo) [inline docs/source/whitepaper.rst:68]: I think this is not necessary after today's discussion. The expiry block number will be checked implicitly by the contracts

File: docs/source/whitepaper.rst [hunk 2]
```diff
@@ -98,6 +99,7 @@ this is our second concern. When filling the request, Bob submits the following
 - target token address
 - recipient address
 - amount
+- request expiry block number
```
Comment (fredo) [inline docs/source/whitepaper.rst:102]: same as above

File: docs/source/whitepaper.rst [hunk 3]
```diff
@@ -84,8 +84,9 @@ explained in the next section.
 In any case, once she received tokens on rollup B, Alice can ignore the rest of the protocol and enjoy her bridged tokens!
 
 If her request cannot be fulfilled by liquidity providers (e.g. amount too high, rollups sequencers offline for some time, ...),
-the `validity period` argument she specified will determine for how long the request will remain open. After this period,
-the request can no longer be filled and she can withdraw her locked tokens on rollup A.
+the `validity period` argument she specified will determine for how long the request will remain open on rollup A. After
+this period, the request can no longer be filled and she can withdraw her locked tokens. On the other hand, the expiry
+block number describes the block number until which the request can be filled on rollup B.
```
Comment (fredo) [inline docs/source/whitepaper.rst:89]: same here

File: docs/source/whitepaper.rst [hunk 4]
```diff
@@ -106,15 +108,17 @@ that can later serve to prove that the request was properly filled.
 ::
 
     request hash = Hash(request ID, source chain ID, target chain ID,
-                        target token address, recipient address, amount)
+                        target token address, recipient address, amount, request expiry block number)
```
Comment (fredo) [inline docs/source/whitepaper.rst:111]: and here

File: docs/source/whitepaper.rst [hunk 5]
```diff
@@ -106,15 +108,17 @@ that can later serve to prove that the request was properly filled.
...
 
 When filling the request, a `fill ID` is also computed, that serves to identify a fill. Detailed information on the
 `fill ID` can be found in section :ref:`fill_id`.
 
+The request expiry block number submitted during the fill prevents Bob to fill the request past its expiry.
```
Comment (fredo) [inline docs/source/whitepaper.rst:116]: and here

File: docs/source/whitepaper.rst [hunk 6]
```diff
@@ -250,27 +254,24 @@ The reason a claimer needs to submit a `fill ID` is to make a statement of when
...
-Upon submitting the claim with a certain `fill ID`, Charles can look for the block with the associated number
-and see whether a fill was correctly made by Bob. Any claim with a different `fill ID` than the generated value upon
-filling the request is considered to be a false claim.
+    fill_id = hash(previous block)
 
-Ideally, the `fill ID` is chosen in a way that it cannot be generated in the future. With the current definition, this
-is not entirely the case and it is subject to being improved in the future.
+When seeing a claim with a certain `fill ID`, observers can verify if a fill with corresponding ID has been made. If they
+know of no fill with this fill ID, they are guaranteed the claim is wrongful, as long as the claimer did not guess the hash
+of a block in the future.
```
Comment (fredo) [inline docs/source/whitepaper.rst:265]:
```suggestion
of a block in the future correctly.
```

**Review by fredo — COMMENTED** [review]
looks good. Please revise the parts about the block number. I think that after today's discussion we don't have that property per request anymore (Or at least we dont need it).

File: docs/source/whitepaper.rst (hunk 1 above)
Comment (cducrest) [reply docs/source/whitepaper.rst]: ok

**Review by fredo — APPROVED** [review]

**Pull request merged**